controllers/automation/policyautomation_controller.go

```diff
@@ -199,6 +205,128 @@ func (r *PolicyAutomationReconciler) Reconcile(
+					// then back to non-compliant during the delay period
+					if delayAfterRunSeconds > 0 && preEventTime.After(originalStartTime) {
+						if now.After(delayUntil) {
+							// The delay period passed so remove the previous event
+							delete(eventMap, clusterName)
+							// Add the cluster name to create a new ansible job
+							trimmedTargetList = append(trimmedTargetList, clusterName)
+						} else {
+							requeueFlag = true
+							// Within the delay period and use the earliest requeueDuration to requeue
+							if (requeueDuration == 0) || (requeueDuration > int(delayUntil.Sub(now)+1)) {
+								requeueDuration = int(delayUntil.Sub(now) + 1)
+							}
+							// keep the event and update eventTime
+							clusterEvent.EventTime = nowStr
+							// new event from compliant to non-compliant
+							eventMap[clusterName] = clusterEvent
+						}
+					} // Otherwise, the policy keeps non-compliant since originalStartTime, do noting
+				} else { // The policy is compliant with the target cluster
+					if delayAfterRunSeconds > 0 && now.Before(delayUntil) {
+						// Within the delay period, keep the event and update eventTime
+						clusterEvent.EventTime = nowStr
+						// new event from non-compliant to compliant
+						eventMap[clusterName] = clusterEvent
+					} else { // No delay period or it is expired, remove the event
+						delete(eventMap, clusterName)
+					}
+				}
+			}
+
+			for _, clusterName := range targetList {
+				if _, ok := eventMap[clusterName]; !ok {
+					// Add the non-compliant clusters without previous automation event
+					trimmedTargetList = append(trimmedTargetList, clusterName)
+				}
+			}
+
+			if len(trimmedTargetList) > 0 {
+				log.Info("Creating An Ansible job", "trimmedTargetList", trimmedTargetList)
+				err = common.CreateAnsibleJob(
+					policyAutomation,
+					r.DynamicClient,
+					string(policyv1beta1.EveryEvent),
+					trimmedTargetList,
+				)
+
+				if err != nil {
+					log.Error(err, "Failed to create the Ansible job")
+
+					return reconcile.Result{}, err
+				}
+
+				automationStartTimeStr := time.Now().UTC().
+					Add(time.Second * time.Duration(1)).Format(time.RFC3339)
+
+				for _, clusterName := range trimmedTargetList {
+					eventMap[clusterName] = policyv1beta1.ClusterEvent{
+						AutomationStartTime: automationStartTimeStr,
+						EventTime:           nowStr,
+					}
+				}
+			} else {
+				log.Info("All clusters are compliant. No new Ansible job. Just update ClustersWithEvent")
+			}
+
+			policyAutomation.Status.ClustersWithEvent = eventMap
+			// use StatusWriter to update status subresource of a Kubernetes object
+			err = r.Status().Update(ctx, policyAutomation, &client.UpdateOptions{})
+			if err != nil {
+				log.Error(err, "Failed to update ClustersWithEvent in policyAutomation status")
+
+				return reconcile.Result{}, err
+			}
+
+			if requeueFlag {
+				log.Info("Requeue for the multi-events during the delay period")
```

Comment:
What do you mean by `multi-events`? Also, could you please log the delay in seconds?